### PR TITLE
Addressing inconsistencies returning as a list of simple haxby datasets.

### PR DESCRIPTION
**Pull request by KamalakerDadi**
changed related documentation.

### CODE REVIEW

**Comment by KamalakerDadi** [comment]
Issue #746 . Addressing inconsistencies giving a list of haxby datasets

**Comment by lesteve** [comment]
Unfortunately you have a merge conflict with Alex's last commit, you need to rebase.

File: nilearn/datasets/func.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 
 def fetch_haxby_simple(data_dir=None, url=None, resume=True, verbose=1):
-    """Download and load an example haxby dataset
+    """Download and load a simple example haxby dataset in a single list.
```
Comment (lesteve) [inline nilearn/datasets/func.py]: in a single list is not important enough to feature in the short explanation (first line of the docstring). Also it is slightly inaccurate.

File: nilearn/datasets/func.py [hunk 2]
```diff
@@ -27,12 +27,12 @@ def fetch_haxby_simple(data_dir=None, url=None, resume=True, verbose=1):
     -------
     data: sklearn.datasets.base.Bunch
         Dictionary-like object, interest attributes are:
-        'func': string.  Path to nifti file with bold data.
-        'session_target': string. Path to text file containing session and
-        target data.
-        'mask': string. Path to nifti mask file.
-        'session': string. Path to text file containing labels (can be used
-        for LeaveOneLabelOut cross validation for example).
+        - 'func': string list.  Path to nifti file with bold data.
```
Comment (lesteve) [inline nilearn/datasets/func.py]: list of string rather than string list.

About adding the hyphen in front of each line, have you checked how it renders in the html API reference documentation? Personally I wouldn't bother doing that in this PR. In case there is a neat improvement in term of html rendering, you would need to do that consistently for all the fetch_* functions.

File: nilearn/datasets/func.py [hunk 3]
```diff
@@ -73,9 +73,9 @@ def fetch_haxby_simple(data_dir=None, url=None, resume=True, verbose=1):
     # There is a common file for the two versions of Haxby
     fdescr = _get_dataset_descr('haxby2001')
 
-    # return the data
-    return Bunch(func=files[1], session_target=files[0], mask=files[2],
-                 conditions_target=files[3], description=fdescr)
+    # return the example data in a list
+    return Bunch(func=[files[1]], session_target=[files[0]], mask=[files[2]],
+                 conditions_target=[files[3]], description=[fdescr])
```
Comment (lesteve) [inline nilearn/datasets/func.py]: I don't think you need brackets around `fdescr` but I may be wrong.

File: nilearn/datasets/func.py [hunk 4]
```diff
@@ -73,9 +73,9 @@ def fetch_haxby_simple(data_dir=None, url=None, resume=True, verbose=1):
     # There is a common file for the two versions of Haxby
     fdescr = _get_dataset_descr('haxby2001')
 
-    # return the data
-    return Bunch(func=files[1], session_target=files[0], mask=files[2],
-                 conditions_target=files[3], description=fdescr)
+    # return the example data in a list
```
Comment (lesteve) [inline nilearn/datasets/func.py]: Slightly inaccurate since you return a Bunch object and not a list. Say something about list being used because the dataset is single-subject.

File: nilearn/datasets/func.py (hunk 3 above)
Comment (AlexandreAbraham) [inline nilearn/datasets/func.py]: No need for `fdescr` nor `mask` which are common to all subjects.

File: nilearn/datasets/func.py (hunk 2 above)
Comment (AlexandreAbraham) [inline nilearn/datasets/func.py]: :+1:

File: nilearn/datasets/func.py (hunk 1 above)
Comment (AlexandreAbraham) [inline nilearn/datasets/func.py]: :+1:

File: nilearn/datasets/func.py [hunk 5]
```diff
@@ -27,12 +27,12 @@ def fetch_haxby_simple(data_dir=None, url=None, resume=True, verbose=1):
...
         Dictionary-like object, interest attributes are:
-        'func': string.  Path to nifti file with bold data.
-        'session_target': string. Path to text file containing session and
+        'func': list of string.  Path to nifti file with bold data.
+        'session_target': list of string. Path to text file containing session and
         target data.
-        'mask': string. Path to nifti mask file.
-        'session': string. Path to text file containing labels (can be used
-        for LeaveOneLabelOut cross validation for example).
+        'mask': list of string. Path to nifti mask file.
```
Comment (AlexandreAbraham) [inline nilearn/datasets/func.py]: `mask` should not be a list.

File: nilearn/datasets/func.py [hunk 6]
```diff
@@ -73,9 +73,10 @@ def fetch_haxby_simple(data_dir=None, url=None, resume=True, verbose=1):
     # There is a common file for the two versions of Haxby
     fdescr = _get_dataset_descr('haxby2001')
 
-    # return the data
-    return Bunch(func=files[1], session_target=files[0], mask=files[2],
-                 conditions_target=files[3], description=fdescr)
+    # return the example data
+    # single subject data in a list and list being used to be consistent
+    return Bunch(func=[files[1]], session_target=[files[0]], mask=[files[2]],
```
Comment (AlexandreAbraham) [inline nilearn/datasets/func.py]: `mask` should not be a list.

**Comment by AlexandreAbraham** [comment]
CircleCI blew up because of memory problem. This is weird since you rebased on master. @lesteve any thoughts on this one?

File: examples/decoding/plot_haxby_anova_svm.py [hunk 7]
```diff
@@ -17,8 +17,8 @@
 print('Functional nifti image (4D) is located at: %s' %
       haxby_dataset.func)
```
Comment (lesteve) [inline examples/decoding/plot_haxby_anova_svm.py]: You probably want to use .func[0] here. I am actually surprised that you didn't have to modify .func -> .func[0] in any of the examples.

**Comment by lesteve** [comment]
> CircleCI blew up because of memory problem. This is weird since you rebased on master. @lesteve any thoughts on this one?

That's weird indeed but I don't have any good suggestions as to why this is happening.

File: examples/decoding/plot_haxby_anova_svm.py (hunk 7 above)
Comment (KamalakerDadi) [inline examples/decoding/plot_haxby_anova_svm.py]: I checked before. It worked not using ```.func[0]```. But, still I can change to be consistent with list. Should I ?

Comment (lesteve) [inline examples/decoding/plot_haxby_anova_svm.py]: haxby.func is a list of 4d nifti filenames, It doesn't seem like a good idea to use that directly and I am actually surprised you don't get an error somewhere...

**Comment by AlexandreAbraham** [comment]
All green, :+1: to merge. @lesteve?

**Comment by lesteve** [comment]
some .func haven't been replaced by .func[0] I think (for example in some print statements). Please double-check @KamalakerDadi.

**Comment by AlexandreAbraham** [comment]
Seems good to go now. @lesteve?

**Comment by lesteve** [comment]
Seems good enough. I'll probably tweak one of the comments post-merge.

**Comment by AlexandreAbraham** [comment]
Cool, thx.
